runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -1017,6 +1041,27 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 #endif /* J9VM_ENV_DATA64 */
 	}
 
+	/**
+	 * Get preserved array layout for the given indexable object
+	 *
+	 * @param forwardedHeader Forwarded header of dstObjectPtr
+	 * @param dstObjectPtr    Pointer to the indexable object whose size is required
+	 */
+	MMINLINE ArrayLayout
+	getPreservedLayout(MM_ForwardedHeader *forwardedHeader, J9IndexableObject *dstObjectPtr)
+	{
+		/* Same as getPreservedClass(forwardedHeader) we don't have access to objectModel so we manually fetch class from forwardedHeader */
+		J9Class* clazz = (J9Class *)((uintptr_t)(forwardedHeader->getPreservedSlot() & J9GC_J9OBJECT_CLAZZ_ADDRESS_MASK));
+
+		/* Trivial check for InlineContiguous. */
+		if (0 != getArraySize(dstObjectPtr)) {
+			return GC_ArrayletObjectModel::InlineContiguous;
+		}
+
+		UDATA dataSizeInBytes = getDataSizeInBytes(clazz, getSizeInElements(dstObjectPtr));
```

Comment:
I realized we have getPreservedIndexableSize (beside class, size is indeed also preserved), so we might not need dstObjectPtr

Reply:
but you don't have object model  :(

Reply:
There's also `getPreservedArrayLayout` but both reside in ObjectModel which is not visible from ArrayletObjectModel. Should we move this functionality to ObjectModel instead?

Reply:
And in fact there is even getPreservedArrayLayout available, but again not easily consumable since it's in base ObjectModel (not ArrayletObjectModel).

Not immediatly clear why are in base, while there are array specific, but not the right time to think about it...